client.go

```diff
@@ -1069,7 +1077,12 @@ func (c *HostClient) Do(req *Request, resp *Response) error {
 		maxAttempts = DefaultMaxIdemponentCallAttempts
 	}
 	attempts := 0
-
+	if req.deadline.IsZero() {
+		timeout := c.ReadTimeout + c.WriteTimeout
+		if timeout != 0 {
+			req.deadline = time.Now().Add(timeout)
```

Comment:
I really don't like having the `deadline` field added to `Request` just for this feature. What is the reason this can't be calculated in `doNonNilReqResp` where it is needed? I would argue that each time we retry because of `MaxIdemponentCallAttempts` should have a new timeout/deadline.